### PR TITLE
Add customRegistrationSuccessScreen type

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1943.

#### Changes proposed in this Pull Request:
- Add customRegistrationSuccessScreen type

### CODE REVIEW

File: src/contexts/AuthUIContextProvider/types.ts [hunk 1]
```diff
@@ -142,6 +142,13 @@ type AuthUIContextProviderProps = {
      * Default: None
      */
     customAccountDetails?: Array<CustomRegistrationForm | null>;
+
+    /**
+     * Custom screen to render when a user successfully completes registration.
+     *
+     * Default: Provides a generic success screen with Name, Email, and Org details.
+     */
+    customRegistrationSuccessScreen?: JSX.Element;
```
Comment (joebochill) [inline src/contexts/AuthUIContextProvider/types.ts:151]: You're probably going to want to do this like we do for the custom header and footer for the login page and let it be a JSX Element or a function that we can pass the navigation object to (for cross-compatibility for react/RN):

```loginHeader?: JSX.Element | ((navigation: any) => JSX.Element);```

That will give us access to the navigation prop in RN too so the user can handle the navigation back to the login screen or wherever they want to go.

Comment (joebochill) [reply src/contexts/AuthUIContextProvider/types.ts:151]: otherwise looks good so far

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
```
Comment (joebochill) [inline CHANGELOG.md:11]: 3.0.0 is published (whoops) on 3/29/2021. This'll need to be a new entry for 3.1.0.

File: docs/API.md [hunk 3]
```diff
@@ -36,6 +36,8 @@ import { AuthUIContextProvider } from '@pxblue/react-native-auth-workflow';
     -   Default: provides a fake phone number
 -   **customAccountDetails**: (optional) _`Array<CustomRegistrationForm | null>`_
     -   Custom forms to add to the user registration process
+-   **customRegistrationSuccessScreen**: (optional) _`JSX.Element`_
```
Comment (joebochill) [inline docs/API.md:39]: Make sure you update the type here to include the function

File: CHANGELOG.md (hunk 2 above)
Comment (joebochill) [reply CHANGELOG.md:11]: And we can bump the number in package.json too.

Comment (daileytj) [reply CHANGELOG.md:11]: Looks like I forgot to add the dates to this one so I'll toss those in there while I'm in here

Comment (joebochill) [reply CHANGELOG.md:11]: I think you may have branched off of an older copy of the repo / dev branch since the dates were in there already in dev.

**Review by joebochill — APPROVED** [review]

**Review by joebochill — COMMENTED** [review]
So after a long night thinking about this in my dreams, I think we should split the two success screens since there may be different requirements for someone who already has an account (e.g., request a role/permissions for the new project) versus a completely new person (already added to the project as part of account creation) — hence why we already make a distinction in the defaults.

So I would suggest splitting this into two props: `registrationSuccessScreen` and `accountAlreadyExistsScreen`.

And I think we should also pass some additional values to the `registrationSuccessScreen` function so that users can use that data (e.g. registrant's name) to build something similar to what we have by default. I would suggest a function signature like:

```
registrationSuccessScreen: JSX.Element | (navigation: any, registrationData: { 
    accountDetails: AccountDetailInformation,
    email: string;
}) => JSX.Element
```

> They will get this information from the `completeRegistration` call that we make, but I think passing the data directly to their component will make their lives easier so they don't have to figure out a way to store that data somewhere in global state and then pass it to their success screen.

File: docs/API.md [hunk 4]
```diff
@@ -18,6 +18,8 @@ import { AuthUIContextProvider } from '@pxblue/react-native-auth-workflow';
 
 #### Available Props
 
+-   **accountAlreadyExistsScreen**: (optional) _`JSX.Element | (navigation: any, registrationData?: RegistrationData) => JSX.Element`_
```
Comment (joebochill) [inline docs/API.md:21]: Correct me if I am wrong, but I don't think we will have any `registrationData` to provide to the user in the 'account already exists' case.

**Review by joebochill — APPROVED** [review]

**Pull request merged**